kernel/timehires.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * timehires.h
+ *
+ * based on code from linux/include/linux/clockcounter.h
+ *
+ * Copyright (C) 2018 Petri Mattila <petri.mattila@unipart.io>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ */
+#ifndef __HIRES_COUNTER_H__
+#define __HIRES_COUNTER_H__
+
+#include <linux/types.h>
+#include <linux/time.h>
+
+#include "kcompat.h"
+
+
+/**
+ * struct hires_counter - hardware abstraction for a free running counter
+ *
+ * @cycle_read:		returns the current cycle value
+ * @time_sync:		timestamp of the syncronization point
+ * @cycle_sync:		cycles count of the syncronization point
+ * @cycle_mask:		bitmask of the running counter bits
+ * @ns_mult1:		multiplier for converting cycles to ns, after sync
+ * @ns_mult2:		multiplier for converting cycles to ns, before sync
+ */
+struct hires_counter {
+	cycle_t (*cycle_read)(const struct hires_counter *);
+	struct timehires time_sync;
+	cycle_t cycle_sync;
+	cycle_t cycle_mask;
+	cycle_t cycle_max;
+	u64 ns_mult1;
+	u64 ns_mult2;
```

Comment:
Could the names "mult1" and "mult2" be changed to reflect what they are used for (i.e. after/before sync)?  Otherwise, it's hard to remember which way round they are when reading the code.